pulsar-client-cpp/tests/BasicEndToEndTest.cc

```diff
@@ -578,8 +578,13 @@ TEST(BasicEndToEndTest, testSinglePartitionRoutingPolicy)
         consumer.receive(m);
         consumer.acknowledgeCumulative(m);
     }
+
+    Promise<bool, Result> closePromise;
+    Result r;
     consumer.close();
-    producer.closeAsync(0);
+    producer.closeAsync(WaitForCallback(closePromise));
```

Comment:
the implementation of `Producer.close()` is exactly the same, so it will exercise the same code path: 

```cpp
Result Producer::close() {
    Promise<bool, Result> promise;
    closeAsync(WaitForCallback(promise));

    Result result;
    promise.getFuture().get(result);
    return result;
}
```